lib/src/key/lq.rs

```diff
@@ -24,6 +25,20 @@ pub fn new<'a>(nd: Uuid, ns: &'a str, db: &'a str, lq: Uuid) -> Lq<'a> {
 	Lq::new(nd, ns, db, lq)
 }
 
+pub fn prefix_nd(nd: &Uuid) -> Vec<u8> {
+	let mut k = [b'/', b'!', b'n', b'd'].to_vec();
+	k.extend_from_slice(nd.as_bytes());
+	k.extend_from_slice(&[0x00]);
+	k
+}
+
+pub fn suffix_nd(nd: &Uuid) -> Vec<u8> {
+	let mut k = [b'/', b'!', b'n', b'd'].to_vec();
+	k.extend_from_slice(nd.as_bytes());
+	k.extend_from_slice(&[0xff]);
+	k
+}
+
```

Comment:
Yes, you are right! But thinking about that now, that means that the users of these structs need to know the hierarchy of the keys.

Say that you don't know the key structure - this will be true for community contributions, new team members, more junior team members/interns, people who don't work with the storage regularly such as from sales engineering or integrations. If we have relative paths declared from the relevant key, they can start navigating based on that information

```
prefix = lq::prefix_nd("some-node-id") // safe, because the API says its safe
prefix = nd::prefix("some-node-id") // potentially not safe, because 'nd' does not imply that it only contains lq, or if it may also contain cf, cluster conf etc.
```

By providing a nd prefix from the lq path, we are saying that that navigation is checked and correct. The other way round requires domain knowledge, which isn't enforced (designs, tests, API).

Reply:
@tobiemh I don't think we established what the outcome we wanted from this was, but will address (if necessary) in sur-253